src/gam/gapi/cloudidentity/groups.py

```diff
@@ -118,43 +118,40 @@ def info():
         for member in members:
             role = get_single_role(member.get('roles', [])).lower()
             email = member.get('memberKey', {}).get('id')
+            member_type = member.get('type', 'USER').lower()
             jc_string = ''
             if showJoinDate:
                 joined = member.get('createTime', 'Unknown')
                 jc_string += f'  joined {joined}'
             if showUpdateDate:
                 updated = member.get('updateTime', 'Unknown')
                 jc_string += f'  updated {updated}'
-            print(
-                f'  {role}: {email}{jc_string}'
-                # f' {member.get("role", ROLE_MEMBER).lower()}: {member.get("email", member["id"])} ({member["type"].lower()})'
-            )
+            print(f'  {role}: {email} ({member_type}){jc_string}')
         print(f'Total {len(members)} users in group')
     elif showMemberTree:
         print(' Member tree:')
-        global cached_group_members
         cached_group_members = {}
-        print_member_tree(ci, name, 2)
+        print_member_tree(ci, name, cached_group_members, 2)
 
 
-def print_member_tree(ci, group_id, spaces):
+def print_member_tree(ci, group_id, cached_group_members, spaces):
     if not group_id in cached_group_members:
         cached_group_members[group_id] = gapi.get_all_pages(ci.groups().memberships(),
                                                             'list',
                                                             'memberships',
                                                             parent=group_id,
+                                                            view='FULL',
                                                             fields='*',
                                                             pageSize=1000)
     for member in cached_group_members[group_id]:
         member_id = member.get('name', '')
         member_id = member_id.split('/')[-1]
-        member_email = member.get('memberKey', {}).get('id')
-        if member_id.isdigit():
-            print(f'{" " * spaces}{member_email} - user')
-        else:
-            print(f'{" " * spaces}{member_email} - group')
-            group_id = group_email_to_id(ci, member_email)
-            print_member_tree(ci, group_id, spaces + 2)
+        role = get_single_role(member.get('roles', [])).lower()
+        email = member.get('memberKey', {}).get('id')
+        member_type = member.get('type', 'USER').lower()
+        print(f'{" " * spaces}{role}: {email} ({member_type})')
```

Comment:
groups can only be members and thus all users that are inherited are members so I suggest we just print "member" on the group and not all the way down the tree.